### PR TITLE
Fix Non Recursive Depth First Search

### PR DESCRIPTION
**Pull request by marcoscannabrava**
### **Describe your change:**

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

- The algorithm was passing tests but was exploring the vertices in a breadth-first approach even though it was using a stack.
- I can add an issue to implement a test for this in the future since the current test only checks the final output but not the search method itself.

### CODE REVIEW

**Comment by cclauss** [comment]
Does the doctring at the top of the file match the new code?

**Comment by spamegg1** [comment]
@marcoscannabrava This file was changed by recent merges (spacing, indentation etc., the content is the same I think) so now it conflicts your older version. You can either rebase, or refork and start over. [This](https://www.w3docs.com/snippets/git/how-to-rebase-git-branch.html) might help with rebase.

@cclauss The docstring at the top is strange. First it says it calls itself recursively, then it says "Instead of recursively calling with a node...", then provides pseudo code that IS recursively calling with a node, then the actual code does not. It's very very confusing. That docstring should be rewritten or removed I think.

**Comment by marcoscannabrava** [comment]
@cclauss & @spamegg1 you're absolutely right. It was an old docstring I hadn't changed. I believe it should be acceptable now.
:+1:

**Comment by spamegg1** [comment]
@marcoscannabrava I see you got rid of the conflict.
Good luck!

File: graphs/depth_first_search.py [hunk 1]
```diff
@@ -1,43 +1,35 @@
...
+       :returns: the trace of the search
+       >>> G = { "A": ["B", "C", "D"], "B": ["A", "D", "E"],
+       ... "C": ["A", "F"], "D": ["B", "D"], "E": ["B", "F"],
+       ... "F": ["C", "E", "G"], "G": ["F"] }
+       >>> start = "A"
+       >>> output_G = list({'A', 'B', 'C', 'D', 'E', 'F', 'G'})
+       >>> all(x in output_G for x in list(depth_first_search(G, "A")))
+       True
+       >>> all(x in output_G for x in list(depth_first_search(G, "G")))
+       True
```
Comment (cclauss) [inline graphs/depth_first_search.py:20]: Why indent these lines?

Comment (marcoscannabrava) [reply graphs/depth_first_search.py]: No particular reason. Just unindented them.

File: graphs/depth_first_search.py [hunk 2]
```diff
@@ -5,19 +5,18 @@
...
+   :returns: the trace of the search
+   >>> G = { "A": ["B", "C", "D"], "B": ["A", "D", "E"],
+   ... "C": ["A", "F"], "D": ["B", "D"], "E": ["B", "F"],
+   ... "F": ["C", "E", "G"], "G": ["F"] }
+   >>> start = "A"
+   >>> output_G = list({'A', 'B', 'C', 'D', 'E', 'F', 'G'})
+   >>> all(x in output_G for x in list(depth_first_search(G, "A")))
+   True
+   >>> all(x in output_G for x in list(depth_first_search(G, "G")))
+   True
```
Comment (cclauss) [inline graphs/depth_first_search.py:19]:
```suggestion
    :param graph: directed graph in dictionary format
    :param vertex: starting vertex as a string
    :returns: the trace of the search
    >>> G = { "A": ["B", "C", "D"], "B": ["A", "D", "E"],
    ... "C": ["A", "F"], "D": ["B", "D"], "E": ["B", "F"],
    ... "F": ["C", "E", "G"], "G": ["F"] }
    >>> start = "A"
    >>> output_G = list({'A', 'B', 'C', 'D', 'E', 'F', 'G'})
    >>> all(x in output_G for x in list(depth_first_search(G, "A")))
    True
    >>> all(x in output_G for x in list(depth_first_search(G, "G")))
    True
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**